### ISSUE TITLE
That will be awesome if DevCPU get that features...

### ISSUE THREAD

**Issue by DankRank** [HerobrinesArmy/DevCPU#13]
That will be awesome if DevCPU get that features:
Debug:
"Step" things,
Disassembly,
Stack View,
Memory View.

Editors for files:
Hex editor,
LEM Color Text editor (with formating features, and symbol table.),
LEM Font editor.

Also, that will be great if last two editors, will can work w/ ".dat" thing.

That is things that really need for me.
I'm working with Floppy Disks, and plain text editor saving text in 8-bit format.
Hex editor is awesome thing to use in that case, so i can do "pHpEpLpLpOp!" w/o any frustration and, show text on screen.
"Decoding" 8-bit to 16-bit LEM color format in program it self is pain for me w/o debugger.
I think i not need to explain why LEM Font editor is needed.

Hope, there will be more friendly UI in next version and Debuger included...

---

There even more features, that i'll like:
Editing Registers "On-Teh-Fly" (ala dcpu-16 studio)
Showing memory addresses where pointing labels (ala studio, again.)

Ok, so why do i need that? Editing Registers, is need when you testing part of code, where for example, "1st function takes 3 (from A); secound (unwiritten ATM) makes A (that is now 3) equal 5; and 3th func takes A again (now w/ 5 in it)"
You can test functions w/ holes in it.

and labels... duh... it's awesome when you got universal code named Device.dasm (that just sets up enirovement for interupts), but you need to test it you need to write helloworld, or something like that. When i finnaly, get that it got no bugs, i'm find minimum 4 errors in HelloWorld. _FAILTURE_ Yeah... I'm compiled code on another IDE (that not works well w/ floppy disks, btw) and, that IDE is showed me errors, and looking at disassembley, i see even more errors.

In Studio (that is on very first specs.), i can see labels' memory addresses, that haves HardWare IDs, and then, using step-by-step thing. looking at disassmebly, and memory, i can see that IDs is writen. But that is, again, uniqe feature of Studio.
## --Your Egor305

EDIT: Just small note about Editing Registers: doo eet in hex. Really, when i get more advanced in DCPU, that was annoying to get that in Decimal at Studio. I think best way, is do switch, so you can edit things in Hex, Dec or Bin.
Also, that switch should have current Register view.

And, also... try get, what. Right, memory edit. :smiley:

**Comment by DankRank** [comment]
Also, keyboard should be able to "leave active". :)

**Comment by HerobrinesArmy** [comment]
The one that was checked in yesterday is only active when the Keyboard pane has the focus.

**Comment by DankRank** [comment]
wait, did key board active if it on "another tab" of active window/sub-window?

**Comment by HerobrinesArmy** [comment]
I'm not sure what you're asking, but when Release 0 Update 0 is made available, it will now behave intuitively. If you click inside the Keyboard pane, it should get the keyboard focus. If you click outside the Keyboard pane it will lose the keyboard focus.

**Comment by DankRank** [comment]
Well, it does it now very good. but, arrow keys is not working. that what i meant:
![willitbeactivelikethat](https://f.cloud.github.com/assets/4050457/345694/54e69a6a-9e29-11e2-9170-a4652f48e3f5.PNG)
Will keyboard get focus like that?

**Comment by HerobrinesArmy** [comment]
No, the keyboard will not get focus like that, though I'm considering making it so you can optionally attach a keyboard to the LEM1802 view, so you can just type in there.

The arrow keys work in the version that's in the repository right now, so they should work properly in Release 0 Update 0.

**Comment by DankRank** [comment]
DOO EET! lol

**Comment by DankRank** [comment]
if you activate a keyboard, and switch to LEM tab, it'll still work.
if you press inside LEM tab, it'll unactivate.
that is bug?

**Comment by HerobrinesArmy** [comment]
That's unintended, but that's kind of awesome. I don't think that works on all platforms.
